lib/SIL/IR/SILFunctionType.cpp

```diff
@@ -2486,7 +2486,7 @@ static CanSILFunctionType getNativeSILFunctionType(
   case SILFunctionType::Representation::Method:
   case SILFunctionType::Representation::Closure:
   case SILFunctionType::Representation::WitnessMethod: {
-    switch (constant ? constant->kind : SILDeclRef::Kind::Func) {
+    switch (origConstant ? origConstant->kind : SILDeclRef::Kind::Func) {
```

Comment:
I believe that except for enum elements, the kind of the requirement SILDeclRef is always going to be equal to the kind of the witness SILDeclRef, no? So this change seems pretty safe for 5.4.

Reply:
In theory, I agree, but I don't know what corner cases lurk in the shadows.